pkg/applier/applier_test.go

```diff
@@ -965,6 +973,39 @@ func TestProcessApplyEvent(t *testing.T) {
 		},
 	}
 	testutil.AssertEqual(t, expectedObjStatusMap, objStatusMap, "expected object status to match")
+	testutil.AssertEqual(t, 1, len(resources.IgnoredObjects()))
+
+	// process failed apply of testObj1 (ignore-mutation object)
+	err = s.processApplyEvent(ctx, formApplyEvent(event.ApplyFailed, testObj1, fmt.Errorf("test error")).ApplyEvent, syncStats.ApplyEvent, objStatusMap, unknownTypeResources, resourceMap, resources)
+	expectedError = ErrorForResourceWithResource(fmt.Errorf("test error"), testObj1ID, testObj1)
+	testutil.AssertEqual(t, expectedError, err, "expected processApplyEvent to error on apply %s", event.ApplyFailed)
+
+	expectedCSE = v1beta1.ConfigSyncError{
+		Code: "2009",
+		ErrorMessage: `KNV2009: failed to apply Test.configsync.test, test-namespace/test-1: test error
+
+source: foo/test.yaml
+namespace: test-namespace
+metadata.name: test-1
+group: configsync.test
+version: v1
+kind: Test
+
+For more information, see https://g.co/cloud/acm-errors#knv2009`,
+		Resources: []v1beta1.ResourceRef{{
+			SourcePath: "foo/test.yaml",
+			Name:       "test-1",
+			Namespace:  "test-namespace",
+			GVK: metav1.GroupVersionKind{
+				Group:   "configsync.test",
+				Version: "v1",
+				Kind:    "Test",
+			},
+		}},
+	}
+	testutil.AssertEqual(t, expectedCSE, err.ToCSE(), "expected CSEs to match")
+	// The object should be removed from the IgnoredObjects cache when ApplyFailed
+	testutil.AssertEqual(t, 0, len(resources.IgnoredObjects()))
```

Comment:
Would be helpful to fix the object, have a successful apply and check the ignoremutation object is added back to the cache.

Reply:
Ack, thanks.